### PR TITLE
Add progress UI when saving blood pressure

### PR DESCRIPTION
**Pull request by Honey14**
https://app.clubhouse.io/simpledotorg/story/496/add-progress-ui-for-saving-blood-pressure-measurements

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt [hunk 1]
```diff
@@ -598,6 +598,7 @@ class BloodPressureEntrySheetLogicTest {
     )
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt:598]: When we add new functionality to a screen migrated from the old legacy architecture, we should add relevant tests in the Mobius style. Can we do this for the `Update` and `ViewRenderer` changes we have introduced in this PR?

Comment (Honey14) [reply app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt:598]: Added tests for `ViewRenderer` and I think we already have tests for saving BP's. Can you check once?

Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt:598]: I meant adding `UpdateSpec` tests for verifying that the state flags are being set correctly.

Comment (Honey14) [reply app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt:598]: Got it

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryUpdateTest.kt [hunk 2]
```diff
@@ -60,4 +65,113 @@ class BloodPressureEntryUpdateTest {
...
+        .then(
+            assertThatNext(
+                hasModel(defaultModel.bloodPressureStateChanged(BloodPressureSaveState.NOT_SAVING_BLOOD_PRESSURE)),
+                hasEffects(SetBpSavedResultAndFinish as BloodPressureEntryEffect)
+            )
+        )
+  }
+
+  @Test
+  fun `when the save button is clicked, then show no change`() {
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryUpdateTest.kt:138]: The name of this test does not indicate that the fact that it is testing the condition where a measurement save operation is ongoing. Please update?

**Pull request merged**